envoy/stream_info/stream_info.h

```diff
@@ -236,6 +236,23 @@ struct UpstreamTiming {
   absl::optional<MonotonicTime> last_upstream_rx_byte_received_;
 };
 
+struct BytesMeterer {
+  uint64_t wireBytesSent() { return wire_bytes_sent_; }
+  uint64_t wireBytesReceived() { return wire_bytes_received_; }
+  uint64_t bodyBytesSent() { return body_bytes_sent_; }
+  uint64_t bodyBytesReceived() { return body_bytes_received_; }
+  void addBodyBytesSent(uint64_t added_bytes) { body_bytes_sent_ += added_bytes; }
+  void addBodyBytesReceived(uint64_t added_bytes) { body_bytes_received_ += added_bytes; }
+  void addWireBytesSent(uint64_t added_bytes) { wire_bytes_sent_ += added_bytes; }
+  void addWireBytesReceived(uint64_t added_bytes) { wire_bytes_received_ += added_bytes; }
+
+private:
+  uint64_t body_bytes_sent_{};
+  uint64_t body_bytes_received_{};
+  uint64_t wire_bytes_sent_{};
+  uint64_t wire_bytes_received_{};
+};
+
```

Comment:
You want a `using BytesMetererSharedPtr = std::shared_ptr<BytesMeterer>` to be consistent with other sharedptrs